pkg/api/validation/cluster_validation.go

```diff
@@ -154,8 +154,17 @@ func ValidateScyllaClusterSpecUpdate(newSpec, oldSpec *scyllav1.ScyllaClusterSpe
 		newRackNames.Insert(rack.Name)
 	}
 	diff := oldRackNames.Difference(newRackNames)
-	if diff.Len() != 0 {
-		allErrs = append(allErrs, field.Forbidden(fldPath.Child("datacenter", "racks"), fmt.Sprintf("racks %v not found, you cannot remove racks from the spec", diff.List())))
+	for _, rackName := range diff.List() {
+		for i, rack := range oldSpec.Datacenter.Racks {
+			if rack.Name != rackName {
+				continue
+			}
+			if rack.Members != 0 {
+				allErrs = append(allErrs, field.Forbidden(fldPath.Child("datacenter", "racks").Index(i), fmt.Sprintf("rack %v not found, racks having members cannot be removed, scale it down to zero members first", rackName)))
+			} else if oldStatus.Racks[rack.Name].Members != 0 {
```

Comment:
Is it gonna panic if the rack isn't present in the status?

Reply:
Map returns the zero value when key is missing, and status has `map[string]RackStatus` type, so zero struct value would be also fine.

Reply:
oh, I forgot we don't use pointers there